lib/heap_sort.rb

```diff
@@ -1,8 +1,23 @@
 
 
 # This method uses a heap to sort an array.
-# Time Complexity:  ?
-# Space Complexity: ?
+# Time Complexity:  I think this is O(n)? It touches every item in the list
+# Space Complexity: O(n)? - creating new sorted array
 def heap_sort(list)
```

Comment:
For time complexity it does touch every item, but removing and adding an item to a heap is O(log n), so this algorithm is O(n log n).  Otherwise well done.